msmbuilder/decomposition/kernel_approximation.py

```diff
@@ -0,0 +1,105 @@
+# Author: Carlos Xavier Hernandez <cxh@stanford.edu>
+# Contributors: Muneeb Sultan <msultan@stanford.edu>, Evan Feinberg <enf@stanford.edu>
+# Copyright (c) 2015, Stanford University and the Authors
+# All rights reserved.
+
+from __future__ import absolute_import
+
+import numpy as np
+from scipy.linalg import svd
+
+from sklearn import kernel_approximation
+from sklearn.metrics.pairwise import pairwise_kernels
+
+from .base import MultiSequenceDecompositionMixin
+
+
+__all__ = ['Nystroem', 'LandmarkNystroem']
+
+
+class Nystroem(MultiSequenceDecompositionMixin, kernel_approximation.Nystroem):
+    __doc__ = kernel_approximation.Nystroem.__doc__
+
+
+class LandmarkNystroem(Nystroem):
+    """Approximate a kernel map using a subset of the training data.
+
+    Constructs an approximate feature map for an arbitrary kernel
+    using a subset of the data as basis.
+    Read more in the :ref:`User Guide <nystroem_kernel_approx>`.
```

Comment:
Not exactly. @evanfeinberg mentioned that this came out of a discussion he had with @rmcgibbo, but it's also pretty much what @schwancr proposed in his ktica manuscript (except we're using the Nystroem approx. here). I did find this [reference](https://papers.nips.cc/paper/5481-fast-prediction-for-large-scale-kernel-machines.pdf) though, which suggests a similar approach (using kmeans centroids as "pseudo-landmark" points).